### ISSUE TITLE
 Cannot identify the given neural network framework, tensorflow-lite

### ISSUE THREAD

**Issue by aaswan** [nnstreamer/nnstreamer#3968]
I am trying to run the following pipeline but getting the error. I am new to nnstreamer. Can someone please help me out here. I have installed the required plugins via 
sudo add-apt-repository ppa:nnstreamer/ppa
sudo apt-get update
sudo apt-get install nnstreamer nnstreamer-caffe2 nnstreamer-tensorflow nnstreamer-tensorflow-lite

**Logs**

test@ws2030077011:~/git/nnstreamer-studio$ gst-launch-1.0 textoverlay name=overlay font-desc="Sans, 26" ! videoconvert ! ximagesink name=img_test uridecodebin uri=file:/home/test/nnstreamer/testing/download.mp4 ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB ! tee name=t_raw t_raw. ! queue ! overlay.video_sink t_raw. ! queue ! videoscale ! video/x-raw,width=224,height=224 ! tensor_converter ! tensor_filter framework=tensorflow-lite model=/home/test/nnstreamer/testing/mobilenet_v1_1.0_224_quant.tflite ! tensor_decoder mode=image_labeling option1=/home/test/nnstreamer/testing/labels.txt ! overlay.text_sink

** (gst-launch-1.0:3105062): WARNING **: 17:09:44.284: Cannot identify the given neural network framework, tensorflow-lite

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_caps_can_intersect: assertion 'GST_IS_CAPS (caps1)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_caps_can_intersect: assertion 'GST_IS_CAPS (caps2)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_pad_template_new: assertion 'caps != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_element_request_compatible_pad: assertion 'GST_IS_PAD_TEMPLATE (templ)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_object_unref: assertion 'object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_caps_can_intersect: assertion 'GST_IS_CAPS (caps2)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_pad_template_new: assertion 'caps != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_element_request_compatible_pad: assertion 'GST_IS_PAD_TEMPLATE (templ)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_object_unref: assertion 'object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_caps_can_intersect: assertion 'GST_IS_CAPS (caps1)' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_unref: assertion 'mini_object != NULL' failed

(gst-launch-1.0:3105062): GStreamer-CRITICAL **: 17:09:44.352: gst_mini_object_ref: assertion 'mini_object != NULL' failed

** (gst-launch-1.0:3105062): CRITICAL **: 17:09:44.352: il_getOutCaps: assertion 'dim[0] > 0 && dim[1] == 1' failed
WARNING: erroneous pipeline: could not link tensorconverter0 to tensorfilter0

**Comment by aaswan** [comment]
I tried with the --gst-debug=3 flag
getting the followinf extra ERROR logs
ERROR           GST_PIPELINE grammar.y:738:gst_parse_perform_link: could not link tensorconverter0 to tensorfilter0
ERROR           GST_PIPELINE grammar.y:738:gst_parse_perform_link: could not link tensorfilter0 to tensordecoder0

**Comment by myungjoo** [comment]
1. PLease write a short and simple pipeline that incurs the same error.
2. Please add URLs to resource files you are using (except for the input video)
3. Please state the execution environments.
4. In the cwd you are working on, please execute ```nnstreamer-check``` utility (from nnstreamer-util.deb) and show its output.

**Comment by aaswan** [comment]
1. I will try to write a shorter pipeline which gives the same error & share
2. 
Model: https://github.com/nnsuite/testcases/raw/master/DeepLearningModels/tensorflow-lite/Mobilenet_v1_1.0_224_quant/mobilenet_v1_1.0_224_quant.tflite

Labels: https://github.com/nnsuite/testcases/raw/master/DeepLearningModels/tensorflow-lite/Mobilenet_v1_1.0_224_quant/labels.txt

Here is the formatted command. 

gst-launch-1.0 --gst-debug=3 textoverlay name=overlay font-desc="Sans, 26" ! videoconvert ! \
ximagesink name=img_test uridecodebin \
uri=file:/home/test/nnstreamer/testing/download.mp4 ! \
videoconvert ! videoscale ! \
video/x-raw,width=640,height=480,format=RGB ! \
tee name=t_raw t_raw. ! queue ! \
overlay.video_sink t_raw. ! queue ! videoscale ! \
video/x-raw,width=224,height=224 ! tensor_converter ! \
tensor_filter framework=tensorflow-lite \
model=/home/test/nnstreamer/testing/mobilenet_v1_1.0_224_quant.tflite ! \
tensor_decoder mode=image_labeling \
option1=/home/test/nnstreamer/testing/labels.txt ! overlay.text_sink

 

3. We are working on Ubuntu 20.04 environment. 


4. test@ws2030077011:~/git/nnstreamer-studio$ nnstreamer-check
NNStreamer version: 2.3.0
           loaded : TRUE
           path   : /usr/lib/x86_64-linux-gnu/gstreamer-1.0/libnnstreamer.so

NNStreamer configuration:
============================================================
Configuration Loaded: TRUE
Configuration file path: /etc/nnstreamer.ini
    Candidates: envvar(NNSTREAMER_CONF): (null)
                build-config: /etc/nnstreamer.ini
                hard-coded: /etc/nnstreamer.ini
[Common]
  Enable envvar: TRUE
  Enable sym-linked subplugins: TRUE
[Filter]
  Filter paths from .ini: /usr/lib/nnstreamer/filters/
             from envvar: (null)
         from hard-coded: /usr/lib/nnstreamer/filters/

============================================================

NNStreamer registered sub-plugins:
============================================================

[Filter]
  caffe2
    - No custom property found
  cpp
    - No custom property found

[Decoder]
  bounding_boxes
    - No custom property found
  image_segment
    - No custom property found
  octet_stream
    - No custom property found
  pose_estimation
    - No custom property found
  direct_video
    - No custom property found
  image_labeling
    - No custom property found

[Conterver]

============================================================

I can see that no sub-plugins are registered. Even though i have installed 
sudo apt-get install nnstreamer nnstreamer-caffe2 nnstreamer-tensorflow nnstreamer-tensorflow-lite

**Comment by aaswan** [comment]
Thanks @myungjoo for the quick resolution. It works now.

"it appears that we have obsoleted tensorflow1 and have not removed tensorflow1 subpackages from PPA.
Could you please try nnstreamer-tensorflow2-lite package in Ubuntu
"
